csharp/LogicalColumnStream.cs

```diff
@@ -25,6 +24,23 @@ public virtual void Dispose()
             Source.Dispose();
         }
 
+        protected static bool RepLevelsRequired(Type type)
```

Comment:
I wonder if these should be `private protected` methods. The `LogicalColumnStream` class is public due to `LogicalColumnReader` etc being public, but we don't actually expect users to implement their own classes deriving from this.